src/game_events/handlers.cpp

```diff
@@ -241,10 +242,24 @@ struct filter_attack : public event_filter {
 	{
 		const unit_map& units = resources::gameboard->units();
 		const auto& loc = first_ ? event_info.loc1 : event_info.loc2;
-		auto unit = units.find(loc);
-		if(unit != units.end() && loc.matches_unit(unit)) {
-			auto attack = event_info.data.optional_child(first_ ? "first" : "second");
-			return swf_.empty() || matches_special_filter(attack, swf_);
+		const auto& loc_d = first_ ? event_info.loc2 : event_info.loc1;
+		auto unit_a = units.find(loc);
+		auto unit_d = units.find(loc_d);
+		if(unit_a != units.end() && loc.matches_unit(unit_a)) {
+			const auto u = unit_a->shared_from_this();
+			auto temp_weapon = event_info.data.optional_child(first_ ? "first" : "second");
+			const_attack_ptr attack = std::make_shared<const attack_type>(*temp_weapon);
```

Comment:
This is wrong. Either use `mandatory_child` or check if the child actually exists. I'm not sure which is correct here, but using `optional_child` and then assuming it exists is definitely not correct.

Reply:
Oh, didn't notice the conflict with #7497.